daft/table/table_io.py

```diff
@@ -212,6 +213,41 @@ def read_parquet(
     return _cast_table_to_schema(MicroPartition.from_arrow(table), read_options=read_options, schema=schema)
 
 
+def read_sql(
+    sql: str,
+    url: str,
+    schema: Schema,
+    sql_options: TableReadSQLOptions = TableReadSQLOptions(),
+    read_options: TableReadOptions = TableReadOptions(),
+) -> MicroPartition:
+    """Reads a MicroPartition from a SQL query
+
+    Args:
+        sql (str): SQL query to execute
+        url (str): URL to the database
+        schema (Schema): Daft schema to read the SQL query into
+
+    Returns:
+        MicroPartition: MicroPartition from SQL query
+    """
+    pa_table = SQLReader(
+        sql,
+        url,
+        # TODO(Colin): Enable pushdowns
+        projection=None,
+        predicate=None,
```

Comment:
I left it out so that we could enable the predicate pushdown into SQL query in a following PR as a perf improvement. But I'm also good to include them in this one, the tests should all still pass!